### DISCUSSION TITLE
Vim Changelog

### DISCUSSION THREAD

**Discussion by igbanam** [vim/vim#15074] [Ideas]
Vim's been seeing some healthy cadence with releases these days. It would be nice to know what changes a 9.1450 -> 9.1500 bump brings. 

How best do you think this can be achieved?

I've looked through the tag information, but these tie directly to Github issues. It would be nice to have a changelog somewhat which hints to what one should expect in a release.

**Comment by chrisbra** [comment]
There used to be https://www.arp242.net/vimlog/ but it hasn'bt neen updated in a while. ping @arp242 
Most notable changes are recorded in [version9.txt](https://github.com/vim/vim/blob/master/runtime/doc/version9.txt). With git you can directly compare the changes like this:
```
git diff v9.1.0450...v9.1.0500 runtime/doc/version9.txt
```
I am sure, github supports something similar with the compare feature, I just don't know how to specify the URL for this.

**Reply by benknoble** [reply]
A more detailed summary would be `git log v9.1.0450...v9.1.0500`, optionally with `--oneline`, etc. Using `--ignore-all-space` with the diff is helpful for that particular series, too.

**Reply by arp242** [reply]
The way I is by look for changes is by looking only at the `doc` directory, reasoning that if something is user-visible it also has a documentation change. So that's: `git log v9.1.0450...v9.1.0500 doc`.

**Comment by damnskippy** [comment]
@chrisbra something like this maybe is what you were saying?  https://github.com/vim/vim/compare/v9.1.0510...v9.1.0513

But the arp242 list was useful in that it listed only the "most useful and notable ones" and not every change which is too much.

**Reply by chrisbra** [reply]
no, I meant to specifically compare version9.txt, similar to what the mentioned git command does. 

> but the arp242 list was useful in that it listed only the "most useful and notable ones" and not every change which is too much.

It's open source, you can create a PR with notable changes listed

**Reply by arp242** [reply]
> It's open source, you can create a PR with notable changes listed

Pretty much that, yes. I haven't updated it in a while because I haven't really felt like it and worked on other stuff. It's not hard, but a little bit boring and tedious.

**Comment by balki** [comment]
Can we create a `CHANGELOG.md` file vim repo. This way pull requests can also add a brief changelog under right section and once a month they can be cleaned up. I imagine it would look like this:

## July 2024

### Breaking Changes
...

### Core Features
#### vim9script
1. ...
2. ...
#### Other

### Runtime changes
#### New filetype supported
...
#### Improved builtin plugins
...
#### Translations
...
### Bugfixes
...

## June 2024
...

---
Looking at all the commits with git log is too tedious and most of them are not applicable to everyone.

**Reply by arp242** [reply]
> pull requests can also add a brief changelog 

This is going to merge conflict like crazy.

I proposed adding a more useful changelog in 2021: #8622, but you really don't want to do it by PR.

**Reply by igbanam** [reply]
It sure would be nice to have a changelog which ships with Vim. What would be the best way to do this? 

Mustn't be per PR. Could be per release. Where the tagged commit would only have changes to the changelog file. This changelog file would comb all commits going into the release and fill in the format above — something like this.

**Reply by arp242** [reply]
Be the change you want to see: `git log doc` and send PR to my repo. That would be a good start.

And yes, this is boring work. But don't expect others to do the work for you.

**Reply by chrisbra** [reply]
whats wrong with inspecting version9.txt?
